DATAJDBC-356 - Support for reading large resultsets

denniseffing opened the pull request:
Support for reading large result sets using the new `queryForStream` methods in Spring Framework 5.3.

schauder commented on spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/AbstractJdbcQuery.java:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 the original author or authors.
+ * Copyright 2021 the original author or authors.
```
I'm just mentioning this because you obviously put quite some effort in adhering to all the rules (Thanks for that).

Copyright should be `2000-2021`.

schauder commented on spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/AbstractJdbcQuery.java:
```diff
@@ -89,10 +91,14 @@ public JdbcQueryMethod getQueryMethod() {
 			return extractor != null ? getQueryExecution(extractor) : collectionQuery(rowMapper);
 		}
 
+		if (queryMethod.isStreamQuery()) {
```
I don't think we should ignore a configured `ResultSetExtractor`. This would make the actual result in terms of instances returned depend on the container (`Stream` vs. `Collection`).

Instead we should fall back to the existing behaviour for `ResultSetExtractor`.

denniseffing replied:
Sorry, I didn't have the time to get back to you on this, but I'll finish this up in the upcoming weeks. Looking forward to getting this merged. 🙂

schauder reviewed (changes requested):
Excellent PR.
I have just the single request to change the behaviour in the presence of a `ResultSetExtractor`.

As another side note: The "Closes" and "Original pull request" should reference the github issue number and the number of this PR.
But I'm happy to add these when merging

denniseffing commented:
@schauder I finally got to finish this up! 
I rebased the PR onto the current main branch and incorporated your suggestion to fall back to the existing behavior when a `ResultSetExtractor` is specified. Please let me know if you require any additional changes.

schauder commented:
Thanks.
I added some documentation and merged this PR.
